apps/src/componentLibrary/tabs/_Tab.tsx

```diff
@@ -0,0 +1,97 @@
+import classNames from 'classnames';
+import React, {useCallback} from 'react';
+
+import FontAwesomeV6Icon, {
+  FontAwesomeV6IconProps,
+} from '@cdo/apps/componentLibrary/fontAwesomeV6Icon';
+import moduleStyles from '@cdo/apps/componentLibrary/tabs/tabs.module.scss';
+
+export interface TabModel {
+  /** Unique value of the tab */
+  value: string;
+  /** Tab text */
+  text?: string;
+  /** Icon left from label */
+  iconLeft?: FontAwesomeV6IconProps;
+  /** Icon right from label */
+  iconRight?: FontAwesomeV6IconProps;
+  /** Whether button should be icon only */
+  isIconOnly?: boolean;
+  /** Tab icon */
+  icon?: FontAwesomeV6IconProps;
+  /** Tab content */
+  tabContent: React.ReactNode;
+  /** Is tab disabled */
+  disabled?: boolean;
+}
+
+interface TabsProps extends TabModel {
+  /** Is tab selected */
+  isSelected: boolean;
+  /** Tab onClick handler */
+  onClick: (value: string) => void;
+  /** The ID of the panel element that this tab controls */
+  tabPanelId: string;
+  /** The ID of the button element (_Tab) */
+  tabButtonId: string;
+}
+
+const checkTabForErrors = (
+  isIconOnly: boolean,
+  icon: FontAwesomeV6IconProps | undefined,
+  text: string | undefined
+) => {
+  if (isIconOnly && !icon) {
+    throw new Error('IconOnly tabs must have an icon');
```

Comment:
Is throwing an error too aggressive here? I saw we do this in some other components too, so maybe it's a pattern we're adopting.

Reply:
Yep, this here is preventing us from rendering impossible combinations of icons/text. Considering it to be something ike typescript type check and have same weigth. (Failing types in TS casuses js erors)